app/models.py

```diff
@@ -1598,7 +1604,11 @@ def __repr__(self):
     def to_dict(self):
         """Serialize this object to a Python dictionary."""
         if sa.inspection.inspect(self).expired:
-            DBSession().refresh(self)
+            try:
+                self = DBSession().merge(self)
+                DBSession().refresh(self)
+            except sa.orm.exc.ObjectDeletedError:
```

Comment:
Perhaps this error should just propagate.

Reply:
I think I added this because in some cases the object is deleted but you don't want an error (like when tearing down the fixtures).

Reply:
This means returning different representations of the object at different times, which sounds like a recipe for disaster.

Reply:
Yeah that's a good point. Let me try it and see how disruptive it is.

Reply:
I mean how disruptive it is to let the exception propagate...

Reply:
Ah lol